Framework/PythonInterface/mantid/FunctionWrapper.py

```diff
@@ -0,0 +1,258 @@
+from mantid.simpleapi import FunctionFactory
+
+class FunctionWrapper:
+# Wrapper class for Fitting Function 
+  def __init__ (self, name, **kwargs):
+    if not isinstance(name, str):
+       self.fun = name
+    else:
+       self.fun = FunctionFactory.createFunction(name)
+    
+       # Deal with attributes first
+#       for key in kwargs:
+#          if(self.fun.hasAttribute(key)):
+#              self.fun.storeAttributeValue(key, kwargs[key])
+            
+       # Then deal with parameters        
+       for key in kwargs:
+          self.fun.setParameter(key, kwargs[key])
+             
+  def __getitem__ (self, name):
+      return self.fun.getParameterValue(name)
+      
+  def __setitem__ (self, name, value):
+      self.fun.setParameter(name, value)
+      
+  def __str__ (self):
+      return self.fun.__str__()
+                 
+  def __add__ (self, other):
+      if(isinstance(self,CompositeFunctionWrapper)):
+        sum = self
```

Comment:
Because I thought it might prevent undesired modification of self, but I'm not sure of this.